tui/network.py

```diff
@@ -9,128 +9,215 @@
-        static_rb.setCallback(dhcp_change, ())
-        if defaults.ipaddr:
-            ip_field.set(defaults.ipaddr)
-        if defaults.netmask:
-            subnet_field.set(defaults.netmask)
-        if defaults.gateway:
-            gateway_field.set(defaults.gateway)
-        if defaults.dns:
-            dns_field.set(defaults.dns[0])
-    else:
-        dhcp_rb = SingleRadioButton("Automatic configuration (DHCP)", None, 1)
+    def choose_primary_address_type(nic):
+        gf = GridFormHelp(tui.screen, 'Networking', 'Address type', 1, 8)
+        txt = "Choose an address type for %s (%s)" % (nic.name, nic.hwaddr)
+        text = TextboxReflowed(45, txt)
+
+        b = [("Ok", "ok"), ("Back", "back")]
+        buttons = ButtonBar(tui.screen, b)
+
+        # IPv4 by default
+        ipv4_rb = SingleRadioButton("IPv4", None, 1)
+        ipv6_rb = SingleRadioButton("IPv6", ipv4_rb, 0)
+        dual_rb = SingleRadioButton("Dual stack (IPv4 primary)", ipv6_rb, 0)
+
+        gf.add(text, 0, 0, padding=(0, 0, 0, 1))
+        gf.add(ipv4_rb, 0, 2, anchorLeft=True)
+        gf.add(ipv6_rb, 0, 3, anchorLeft=True)
+        gf.add(dual_rb, 0, 4, anchorLeft=True)
+        gf.add(buttons, 0, 5, growx=1)
+
+        loop = True
+        direction = LEFT_BACKWARDS
+        address_type = None
+        while loop:
+            result = gf.run()
+            if buttons.buttonPressed(result) == 'back':
+                loop = False
+            elif buttons.buttonPressed(result) == 'ok':
+                value = None
+                if ipv4_rb.selected():
+                    value = "ipv4"
+                elif ipv6_rb.selected():
+                    value = "ipv6"
+                elif dual_rb.selected():
+                    value = "dual"
+                loop = False
+                direction = RIGHT_FORWARDS
+                address_type = value
+
+        tui.screen.popWindow()
+        return direction, address_type
+
+    def get_ip_configuration(nic, txt, defaults, include_dns, iface_class):
+        def use_vlan_cb_change():
+            vlan_field.setFlags(FLAG_DISABLED, vlan_cb.value())
+
+        def dhcp_change():
+            for x in [ ip_field, gateway_field, subnet_field, dns_field ]:
+                x.setFlags(FLAG_DISABLED, static_rb.selected())
+
+        ipv6 = iface_class == NetInterfaceV6
+
+        gf = GridFormHelp(tui.screen, 'Networking', 'ifconfig', 1, 10)
+        if txt is None:
+            txt = "Configuration for %s (%s)" % (nic.name, nic.hwaddr)
+        text = TextboxReflowed(45, txt)
+        b = [("Ok", "ok"), ("Back", "back")]
+        buttons = ButtonBar(tui.screen, b)
+
+        #TODO? Change size for IPv6? If so which size?
+        ip_field = Entry(16)
+        subnet_field = Entry(16)
+        gateway_field = Entry(16)
+        dns_field = Entry(16)
+        vlan_field = Entry(16)
+
+        static = bool(defaults and (defaults.modev6 if ipv6 else defaults.mode) == NetInterface.Static)
```

Comment:
unnecessary cast to bool?

Reply:
if default is None then static is equal to None as the and is not evaluated.
And it can"t be passed as a bool in the following code.